File: DSA/HR/lonely_integer.java
import java.util.*;

class Result {
    public static int lonelyinteger(List<Integer> a) {
        Collections.sort(a);
        int index = a.size() - 1;
        boolean notFound = true;
        for (int i = 1; i < a.size() && notFound; i += 2) {
            if (!a.get(i - 1).equals(a.get(i))) {
                index = i - 1;
                notFound = false;
            }
        }
        return a.get(index);
    }
}

File: DSA/HR/lonely_integer.cpp
#include <vector>
#include <map>

using namespace std;

int lonelyinteger(vector<int> a)
{
    std::map<int, int> m;

    for (int num : a)
    {
        m[num]++;
    }

    for (const auto &[k, v] : m)
    {
        if (v == 1)
        {
            return k;
        }
    }
    return -1;
}
